pkg/security/security_profile/profile/profile_dir.go

```diff
@@ -433,3 +446,76 @@ func (dp *DirectoryProvider) SendStats(client statsd.ClientInterface) error {
 
 	return nil
 }
+
+func (dp *DirectoryProvider) cleanupLoop(ctx context.Context) {
+	ticker := time.NewTicker(5 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			if err := dp.cleanup(); err != nil {
+				seclog.Errorf("couldn't cleanup profiles: %v", err)
+			}
+		}
+	}
+}
+
+func (dp *DirectoryProvider) cleanup() error {
+	paths, err := dp.listProfiles()
+	if err != nil {
+		return err
+	}
+
+	type wsAndPath struct {
+		selector cgroupModel.WorkloadSelector
+		path     string
+	}
+
+	// read workload selectors from all directory profiles
+	workloadSelectors := make([]wsAndPath, 0)
+	for _, path := range paths {
+		_, workloadSelector, err := readProfile(path)
+		if err != nil {
+			return err
+		}
+
+		workloadSelectors = append(workloadSelectors, wsAndPath{
+			selector: workloadSelector,
+			path:     path,
+		})
+	}
```

Comment:
This feels like the cached profileMapping list is not very useful to me.. I wonder if we could modify the cached one to store "real" workload selector (instead of, somehow a "fake" profile one) and use it to clean up dumps more easily (instead of reconstruct the whole view by re-open every dumps/profiles every 5min). WDYT ?

Reply:
yes, good point. I updated the PR to cleanup a bit the profileMapping, and used it in the cleanup loop (instead of re-reading from disk)